### PR TITLE
Add Quiz to Ad_Hoc, Sorting_Custom

### PR DESCRIPTION
**Pull request by CryoJS**
Added quizzes to the below modules
- Bronze: Ad Hoc
- Silver: Sorting Custom

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [X] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/2_Bronze/Ad_Hoc.mdx [hunk 1]
```diff
@@ -322,3 +329,29 @@ Of course, ad hoc problems can be
...
+    <Quiz.Answer>
+      Being able to quickly recall complex algorithms and data structures.
+      <Quiz.Explanation>
+        Incorrect. Ad hoc solutions may use specific algorithms and data structures, but a majority of the solution relies on observations and understanding the problem.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      Exploring lots of cases and making observations.
+      <Quiz.Explanation>
+        Correct. Trying small cases allows you to better understand the problem and helps you make more observations. Certain observations may be a crucial part of the solution.
```
Comment (TheGamingMousse) [inline content/2_Bronze/Ad_Hoc.mdx:347]: i feel like this isn't really unique to ad hoc problems

and also I guess it is a little subjective (altho I agree with your answer here)

File: content/3_Silver/Sorting_Custom.mdx [hunk 2]
```diff
@@ -1176,3 +1178,50 @@ Many of the problems below may use other Silver concepts, such as
 </Info>
 
 <Problems problems="general" hideSuggestProblemButton />
+
+## Quiz
+
+<Quiz>
+  <Quiz.Question>
+    When are custom comparators used?
```
Comment (TheGamingMousse) [inline content/3_Silver/Sorting_Custom.mdx:1186]: ok i don't really see the point of this quiz

cuz like, a custom comparator is something where you just go like "oh I need to sort these things in a non-predefined way time to write a comparator", and isn't any more complicated than that

File: content/3_Silver/Sorting_Custom.mdx [hunk 3]
```diff
@@ -1176,3 +1178,50 @@ Many of the problems below may use other Silver concepts, such as
...
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      Sorting custom objects or sorting values in non-default order.
+      <Quiz.Explanation>
+        Correct. You will need to use a custom comparator as there is no default option that is able to sort the custom data, or no option to sort in the specific way that you want.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Which input constraints require coordinate compression for the following problem: Given $N$ unique integers, in where each integer is $N_i$ ($1 \leq N_i \leq M$), the solution requires you to create an array where the values of the $N$ integers given are represented by the array's indices.
```
Comment (TheGamingMousse) [inline content/3_Silver/Sorting_Custom.mdx:1207]: idk this problem seems a bit contrived to me, if ur gonna ask a similar question maybe make it a bit more natural

File: content/2_Bronze/Ad_Hoc.mdx (hunk 1 above)
Comment (CryoJS) [reply content/2_Bronze/Ad_Hoc.mdx:347]: I see what you mean. I can make it more clear and include how such specifics lead to a better understanding of the problem (which is unique to ad hoc problems).

File: content/3_Silver/Sorting_Custom.mdx (hunk 3 above)
Comment (CryoJS) [reply content/3_Silver/Sorting_Custom.mdx:1207]: Yep sounds good, I will alter the problem.

File: content/3_Silver/Sorting_Custom.mdx (hunk 2 above)
Comment (CryoJS) [reply content/3_Silver/Sorting_Custom.mdx:1186]: I guess it might be too obvious. I'll alter the question.

**Comment by SansPapyrus683** [comment]
starting to question why we thought quizzes were a good idea in the first place

**Comment by CryoJS** [comment]
> starting to question why we thought quizzes were a good idea in the first place

lmao, I liked the quizzes when I would read the modules
Its more applicable/gamified than reading which can be boring sometimes

**Comment by TheGamingMousse** [comment]
> > starting to question why we thought quizzes were a good idea in the first place
> 
> lmao, I liked the quizzes when I would read the modules Its more applicable/gamified than reading which can be boring sometimes

yeah I personally don't think they're inherently bad, just that it's hard to make a quiz that both has utility and isn't very subjective

**Comment by CryoJS** [comment]
I tried to make these altered questions more application-based! Hope they are good.

**Review by TheGamingMousse — APPROVED** [review]
looks fine to me

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**